app/waf/src/controllers/cert.controller.ts

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright 2019 F5 Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Count,
+  CountSchema,
+  Filter,
+  repository,
+  Where,
+  EntityNotFoundError,
+} from '@loopback/repository';
+import {
+  post,
+  param,
+  get,
+  getFilterSchemaFor,
+  getWhereSchemaFor,
+  patch,
+  del,
+  requestBody,
+  RequestContext,
+  RestBindings,
+  HttpErrors,
+} from '@loopback/rest';
+import {Cert, Adc} from '../models';
+import {CertRepository, AdcRepository} from '../repositories';
+import {Schema, Response, CollectionResponse} from '.';
+import {BaseController} from './base.controller';
+import {inject} from '@loopback/core';
+import {ASGService, ASGManager} from '../services';
+import path = require('path');
+
+const prefix = '/adcaas/v1';
+export class ContentBody {
+  name: string;
+  content: string;
+}
+
+export class CertController extends BaseController {
+  asgMgr: ASGManager;
+
+  constructor(
+    @repository(CertRepository)
+    public certRepository: CertRepository,
+    @repository(AdcRepository)
+    public adcRepository: AdcRepository,
+    @inject(RestBindings.Http.CONTEXT, {optional: true})
+    protected reqCxt: RequestContext,
+    @inject('services.ASGService')
+    public asgService: ASGService,
+  ) {
+    super(reqCxt);
+    this.asgMgr = new ASGManager(this.asgService);
+  }
+
+  @post(prefix + '/certs', {
+    responses: {
+      '200': Schema.response(Cert, 'Successfully create Cert resource'),
+      '400': Schema.badRequest('Invalid Cert resource'),
+      '422': Schema.unprocessableEntity('Unprocessable Cert resource'),
+    },
+  })
+  async create(
+    @requestBody(
+      Schema.createRequest(Cert, 'Cert resource that need to be created'),
+    )
+    cert: Partial<Cert>,
+  ): Promise<Response> {
+    cert.tenantId = await this.tenantId;
+    return new Response(Cert, await this.certRepository.create(cert));
+  }
+
+  @get(prefix + '/certs/count', {
+    responses: {
+      '200': {
+        description: 'Cert model count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async count(
+    @param.query.object('where', getWhereSchemaFor(Cert)) where?: Where,
+  ): Promise<Count> {
+    return await this.certRepository.count(where);
```

Comment:
We may only count the certs who belongs to this user tenant. However, I think other count api also does not implement that. Let's do it later.

Reply:
I will remove this.